servicetest/coredump/test_coredump.py

```diff
@@ -0,0 +1,175 @@
+# BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+# OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+# WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+# ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+# SOFTWARE.
+#
+# For further information see LICENSE
+
+import pytest
+
+import os
+import time
+import signal
+import shutil
+
+from testframework.testhelper import CoreDumpHelper
+from testframework import Container
+from testframework import Capability
+from testframework import DefaultManifest
+
+CURRENT_DIR = os.path.dirname(os.path.realpath(__file__))
+TESTOUTPUT_DIR = CURRENT_DIR + "/testoutput"
+
+# This function is used by the 'agent' fixture to know where the log should be stored
+def logfile_path():
+    return TESTOUTPUT_DIR + "/test.log"
+
+def coredump_path():
+    return "/tmp/coredumps"
+
+# These default values are used to pass various test specific values and
+# configurations to the Container helper. Tests that need to add, remove or
+# update entries can simply base their dict on this one for convenience.
+DATA = {
+    Container.CONFIG: '[{"enableWriteBuffer": false}]',
+    Container.BIND_MOUNT_DIR: "/gateways/app",
+    Container.HOST_PATH: CURRENT_DIR,
+    Container.READONLY: False
+}
+
+# This function is used by the testframework 'testhelper' fixture to know where the
+# testhelper should be made available when running the tests
+def mounted_path_in_host():
+    return CURRENT_DIR
+
+#
+# Gateway config and capability + service manifest for enabling core dumps
+#
+FILE_CONFIG = [{
+    "path-host": coredump_path(),
+    "path-container": coredump_path(),
+    "read-only": False
+}]
+default_cap = Capability("test.cap.default.coredump",
+                        [
+                            { "id": "file", "config": FILE_CONFIG }
+                        ])
+default_manifest = DefaultManifest(TESTOUTPUT_DIR, "default-caps.json", [default_cap])
+
+def service_manifests():
+    """ The agent fixture calls this function when it creates the service manifests
+        that should be used with this test module. The agent fixture expects a list
+        of StandardManifest and/or DefaultManifest objects.
+    """
+    return [default_manifest]
+
+@pytest.fixture
+def create_testoutput_dir(scope="module"):
+    """ Create a directory for the generated test files.
+
+        This directory is ignored by git but it's nice to have
+        somewhere locally to store test output to support
+        troubleshooting etc.
+    """
+    if not os.path.exists(TESTOUTPUT_DIR):
+        os.makedirs(TESTOUTPUT_DIR)
+
+@pytest.fixture
+def create_coredump_dir(scope="module"):
+    """ Create the directories needed for getting the core dumps
+    """
+
+    # Create the core dump directory if it doesn't already exist
+    if not os.path.exists(coredump_path()):
+        os.makedirs(coredump_path())
+
+    yield
```

Comment:
yield returns control to the consuming test, and when the test is done, the fixture does whatever comes after the yield, i.e. cleanup/teardown things.